service/object_creator.rb

```diff
@@ -0,0 +1,74 @@
+module ObjectCreator
```

Comment:
Regarding ObjectCreator at all, I see that each of your conditions have pretty similar logic, let's think how could you improve your code with DRY principles?